src/ctapipe/instrument/optics.py

```diff
@@ -263,3 +269,161 @@ def __repr__(self):
+            Parameters describing the dependency of the radial scale on the distance to the center of the camera
+        az_scale_params : list
+            Parameters describing the dependency of the azimuthal scale scale on the distance to the center of the camera
+        """
+
+        self.asymmetry_params = asymmetry_params
+        self.radial_scale_params = radial_scale_params
+        self.az_scale_params = az_scale_params
+
+        self.update_location(0.0, 0.0)
+
+    def k_func(self, x):
+        return (
+            1
+            - self.asymmetry_params[0] * np.tanh(self.asymmetry_params[1] * x)
+            - self.asymmetry_params[2] * x
+        )
+
+    def sr_func(self, x):
+        return (
+            self.radial_scale_params[0]
+            - self.radial_scale_params[1] * x
+            + self.radial_scale_params[2] * x**2
+            - self.radial_scale_params[3] * x**3
+        )
+
+    def sf_func(self, x):
+        return self.az_scale_params[0] * np.exp(
+            -self.az_scale_params[1] * x
+        ) + self.az_scale_params[2] / (self.az_scale_params[2] + x)
+
+    def pdf(self, r, f):
+        """
+        Calculates the value of the psf at a given location
+
+        Parameters
+        ----------
+        r : float
+            distance to the center of the camera in meters
+        f : float
+            polar angle in radians
+
+        Returns
+        ----------
+        psf : float
+            value of the PSF at the specified location
+        """
+        return laplace_asymmetric.pdf(r, *self.radial_pdf_params) * laplace.pdf(
+            f, *self.azimuthal_pdf_params
+        )
+
+    def update_model_parameters(self, model_params):
+        """
+        Updates the model parameters for the psf
+
+        Parameters
+        ----------
+        model_params : dict
+            dictionary with the model parameters
+            needs to have the keys asymmetry_params, radial_scale_params and az_scale_params
+            The values need to be lists of length 3, 4 and 3 respectively
+        """
+        if not (
+            len(model_params["asymmetry_params"]) == 3
+            and len(model_params["radial_scale_params"]) == 4
+            and len(model_params["az_scale_params"]) == 3
+        ):
+            raise ValueError(
+                "asymmetry_params and az_scale_params needs to have length 3 and radial_scale_params length 4"
+            )
+
+        self.asymmetry_params = model_params["asymmetry_params"]
+        self.radial_scale_params = model_params["radial_scale_params"]
+        self.az_scale_params = model_params["az_scale_params"]
+        k = self.k_func(self.radial_pdf_params[1])
+        sr = self.sr_func(self.radial_pdf_params[1])
+        sf = self.sf_func(self.radial_pdf_params[1])
+        self.radial_pdf_params = (k, self.radial_pdf_params[1], sr)
+        self.azimuthal_pdf_params = (self.azimuthal_pdf_params[0], sf)
+
+    def update_location(self, r, f):
```

Comment:
Why do we need an "update" function? Do we have to pre-compute expensive operations? 

Why is evaluating the PSF not enough?

Reply:
I have right now 2 use cases for this PSF model: processing variance images (e.g. finding stars in variance images) and fitting the PSF parameters based on stars in variance images. for the former i need to evaluate it the PSF at different locations for each possible star location, in the latter i also need to vary the psf parameters for fitting. It made sense to me to have these 2 functions available to modify these parameters. 

Also just to clarify, update location is where the "point" in point spread function is, as in if you want the PSF for a point source at r = 1m and f = 1.0rad these coordinates will need to be put in this function.

Reply:
i will try and make the documentation more clear.

Reply:
The question is why this point needs to be state on the class and cannot just be evaluated.

Reply:
I believe this is a remnant of the debugging code I used during the development... :)
@ctoennis, we don't need this step to be factored out like this anymore.

Reply:
ok, i'll put it all together then. I am having a little issue with the docustrings i'm sorting out now. I'll get to it after that.